site/docs/v1/tech/identity-providers/openid-connect/cognito.adoc

```diff
@@ -0,0 +1,135 @@
+
+You will first need to log in to link:https://aws.amazon.com[AWS].
+
+Once logged in, search for "Cognito" in the main search field, and select the "Cognito" service.
+
+- Choose the region you'd like the user pool to reside in from the top right dropdown.
+- Click "Create a user pool".
+- Select the pool sign-in options for Step 1. In this example, we are using email.
+- Review the configurations for Step 2 up to Step 4 to make sure they conform to your needs.
+- Give the pool a name in Step 5.
+- Make sure the checkbox to use hosted authentication pages is checked.
+- Choose a domain for the user pool.
+
+=== Creating a User Pool App Client
+
+To enable FusionAuth to access the user pool, we need to set up an app client on Cognito. In your user pool:
+
+- Under "Initial app client" on Step 5, set the app type to confidential.
+- Give the client a name. We’ve used "FusionClient".
+- Set the "Callback URL" to `<YOUR_FUSIONAUTH_URL>/oauth2/callback` under allowed callback URLs.
+- Under "Authentication flows" in the "Advanced app client settings" make sure the "ALLOW_USER_PASSWORD_AUTH" auth flow is selected.
+- Check the box for your app name under "Identity Providers".
+- Select "Authorization code grant" under "OAuth 2.0 grant types".
+- Under "OpenID Connect scopes" select "OpenID".
+- Review the attribute read and write permissions and then click "Next" to review the user pool configuration details.
+- Scroll down and click "Create user pool".
+- Once the user pool is created, click on the user pool's name and scroll to the "App clients and analytics section".
+- Open the created app client and record both the "Client ID" and "Client Secret" which can be revealed by toggling the "Show client secret" button.
+- The user pool and app client are now created and the next step is adding a test user.
+- Open the user pool and under the users tab click "Create user".
+
+
+
+=== Configure a New FusionAuth OpenID Connect Identity Provider
+
+There is no pre-configured connector for Cognito in FusionAuth. The generic "OpenID" can be used though, as Cognito supports the standard OpenId Connect protocols.
+
+Navigate to your FusionAuth instance. Select "Settings" from the sidebar, and then "Identity Providers".
+
+Select "Add OpenID Connect" from the "Add" dropdown at the top right of the page.
+
+image::identity-providers/cognito/cognito-choose-openid.png[Create a new OpenID integration, width=1200, role=bottom-cropped]
+
+- Provide a "Name", like "Cognito".
+- Set "Client Id" to the "App Client Id" recorded when creating the app client on Cognito.
+- Select "HTTP Basic Authentication" for the "Client Authentication" field.
+- Set "Client secret" to the app client secret recorded when creating the app client on Cognito.
+
+To retrieve the values for the various "endpoints" fields, visit:
```

Comment:
Could you not use `Discover endpoints`? If that didn't work, please say "uncheck 'Discover endpoints'" or similar.